src/linter.js

```diff
@@ -671,4 +689,32 @@ export default class Linter {
       });
     }
   }
+
+  /*
+    Analyze all HTML files in the package to figure out
+    if we should enable `sourceType: module` instead of parsing the file
+    as a regular script for a particular file.
+  */
+  async _extractJsModules() {
```

Comment:
@EnTeQuAk I'm wondering if parsing the html files is really worth it, especially given that it seems that we are always going to resort to "try to re-parse the js file if an error related to ES6 import and export syntax has been raised while parsing the file in the default parsing mode" (at least based on the current implementation of the `verifyParserOptionsAndGetSourceCode` method), and so I'm not really sure that the complexity it adds to the change is properly balanced by useful issues that this may be able to catch (and it is clear that is going to miss a good chunk of them, e.g. tag scripts injected at runtime, es6 modules imported by other modules, and the dynamic imports once officially supported).

Reply:
I think I agree, I'm going to remove it. It does set the proper `sourceType` upfront but at the cost of parsing all html files in the add-on, even though there may not be any es6 modules anywhere.

It's probably best to add some eslintrc fetching in a later addition instead where people can simply declare their add-on as "uses es6 modules" and we're going to use that information.